### ISSUE TITLE
In-game storyline/ship-unlock hints

### ISSUE THREAD

**Issue by 1010todd** [endless-sky/endless-sky#7961]
### Problem Description

Currently player have very little chance of player knowing how to get obtainable human licenses like City-Ship License or find the Emerald Sword on their own, you have to just happen to be doing those specific jobs and/or missions offered in limited area to get it without googling. I think there should be random spaceport missions that allow you to talk to someone who may already own such license and have them give you hints on how you can get it or in Emerald Sword case, hint that bring bands around might give you something cool.

### Related Issue Links

https://github.com/endless-sky/endless-sky/issues/3663
https://github.com/endless-sky/endless-sky/issues/4907 (Closely related but I think it's too specific on prophet idea in the opening+heavily thumbdowned it might deter people from reading further.)
https://github.com/endless-sky/endless-sky/pull/7849
https://github.com/endless-sky/endless-sky/issues/7583
https://github.com/endless-sky/endless-sky/pull/7958

https://discord.com/channels/251118043411775489/266345072554016768/980237805160636446 ("Rumor" page for tracking mission hints?)
https://discord.com/channels/251118043411775489/266345072554016768/800475764445151252 (Archivist giving lore info and mission hints.)

### Desired Solution

For example, you could meet a Bactrian owner when you land somewhere and you could either ask or he happened to be telling a story and mention how he was running some odd jobs that apparently was offered just around The Deep, then ran into some Deep Security officer and ended up doing a few jobs for him, then .... It doesn't have to be exact but just enough for player to notice something when they ended up getting those jobs/missions as well. And then possibly do the same for Militia license/joining FW where the captain met some militia officer pointing gun at their head when trying to recruit them. Similarly for Emerald Sword maybe a captain could tell you about perks and benefits of helping out some famous bands, how one of the band member told him about some outlandish story he didn't believe at first but once he tried it appeared to be true and these bands might be up to something. Kestrel could possibly just be someone from inner circle of Tarazed telling you they're looking for skilled captain to test their ship (if we want to make it easier to find that is, I don't see as many people complaining about it though).

To make it the most effective these hinting missions should be offered anywhere within human space so we don't hit the same block as storymissions where player have to be wandering around that area to get it offered.

### Alternative Approaches

Putting these hints into spaceport news or merchant hails. This will likely be much less effective and they're often just random chatters that too often doesn't lead anywhere. A mission with proper conversation panel would make it look more important.
(https://discord.com/channels/251118043411775489/251118265965871104/756433159955939328 , related; idea of giving spaceport news weight)

**Comment by Zitchas** [comment]
1. Yes, definitely. This sort of content would be great.
2. That being said, regarding the alternate approaches, this is one case where they should not be "alternates" but rather "one of many approaches used. As in, we should be doing *all* of these methods, not just one of them. Hails and starport news stories are as meaningful as we make them. They could, theoretically, be filled with hints about upcoming missions, random stuff like price spikes (or markets bottoming out) somewhere, or other useful stuff. The stuff we have right now is mostly filler, flavor. If we could add meaningful stuff to the point where roughly a third of all starport news and hails actually had something of gameplay value in them, that'd shift the perception of them immensely.

TIMBR is a sore point for me, I think. So much stuff depends on it, and yet it is sized such that people focusing on transport stuff will get it super early (maybe even in their shuttle if they optimize for it), where as cargo and combat oriented players aren't likely to see it until they've spent several million on ships and stuff, assuming they ever do see it. Which basically gates people flying these two perfectly valid routes from a lot of pre-campaign content. If there were alternate routes into the Deep missions or Rim archeology, it'd be fine. Good, even. But I haven't even heard a hint of anyone doing so.

So right now, one of the most valuable hints would be something like "I heard that ensuring your ship has at least 8 free bunks is a great way to pick up celebrity customers..."

<details>
<summary>Expansion on my disatisfaction with TIMBR</summary>

In my recent playthrough, my most annoying discovery about the TIMBR missions is that they won't show at all, not even a blocked message or something, unless I have 8 bunks free. Considering how many missions depend on TIMBR, most of which feel like early game content (and work very well as pre-campaign content).... Well, how many starter ships have 8 free bunks? None. OK, how many "likely first upgrade" ships have 8 free bunks? Only a single one, the heavy shuttle.

If you are going the interceptor/combat route, there's no likely candidates for having 8 free bunks until one reaches medium warships. 

And for players focusing on cargo, there isn't a ship that natively has 8 free bunks until one reaches the Argosy at 2.7M credits. Except that a real cargo focused player probably went with a freighter, which has better cargo than a Argosy and is only 1.4M, and after that the next actual upgrade is the Hauler II at 3.7M, which is when they'll finally have enough free bunks to pick up TIMBR, 

In contrast, a veteran player who knows what they are doing can get 13 free bunks on their starting shuttle without compromising anything, and the heavy shuttle comes stock with 9 free.

Basically, TIMBR is a mission that is almost exclusively for Transport pilots or people who know what they're doing, and due to its size, is likely to not be run into by people going the cargo or combat routes until they're already getting busy with other things. 

</summary>

**Comment by Hecter94** [comment]
> In my recent playthrough, my most annoying discovery about the TIMBR missions is that they won't show at all, not even a blocked message or something, unless I have 8 bunks free.

One option would be to allow the mission to offer, but only allow the player to accept if they have eight bunks free. 
As in, "Can we pay you for transport?"
"Sorry, I don't have enough bunks for eight people", and then defer the mission allowing it to be offered again at a future date.

**Comment by warp-core** [comment]
> > In my recent playthrough, my most annoying discovery about the TIMBR missions is that they won't show at all, not even a blocked message or something, unless I have 8 bunks free.
> 
> One option would be to allow the mission to offer, but only allow the player to accept if they have eight bunks free. As in, "Can we pay you for transport?" "Sorry, I don't have enough bunks for eight people", and then defer the mission allowing it to be offered again at a future date.

There isn't currently a blocked message, we could add something like: "As you navigate the spaceport, you overhear a woman looking for passage for <however many> people and <some> cargo. Unfortunately, you do not currently have that much space available and so are unable to take on the job."

If a player were at a location with an outfitter and/or shipyard when they received such a message, I believe it would actually be possible for them to make adjustments to increase their capacity (assuming they have the money, etc) and then return to the spaceport to actually receive the mission.

For more complexity, we could add support for an `on blocked` mission action.

**Comment by Zitchas** [comment]
The TIMBR one is really the egregious stand out here: Who would think that bumping up the number of bunks on your interceptor so you can give a band a lift not once, but several widely disconnected times, was an essential component to not just one, but multiple story lines? Not to mention the fact that the TIMBR missions are really spread out, and there's not much info to direct the player to the next one.
